src/encoded/static/components/award.js

```diff
@@ -474,67 +1319,177 @@ const ChartRenderer = (props) => {
             searchData[chartCategory].statuses = (statusFacet && statusFacet.terms && statusFacet.terms.length) ? statusFacet.terms : [];
         }
     });
+    // If there are experiements, then the corresponding species are added to the array of species
```

Comment:
This code is very repetitive, I think the common elements should be refactored into a function or class.